### ISSUE TITLE
[Feature Request] minimum enclosing rectangle

### ISSUE THREAD

**Issue by zccyman** [kornia/kornia#269]
How do I get the minimum enclosing rectangle like  using the OpenCV function minAreaRect?

**Comment by edgarriba** [comment]
@zccyman nice feature request ! I open here a discussion with things to consider:
- API
```python
def min_area_rect(points: torch.Tensor) -> torch.Tensor
```
where points is a tensor with shape (B, N, 2) and should return a rotated rectangle which I guess can be parametrized by: [x_center, y_center, height, width, orientation], so return shape is (B, N, 5).

- I include here a reference for the implementation
https://gis.stackexchange.com/questions/22895/finding-minimum-area-rectangle-for-given-points

- I open here a discussion whether we should represent this primitive shapes like line, point, rectangle, oriented_rectangle, circle with OOT (Oriented Object Tensor). A topic that was recently discussed during the pytorch devcon with @jph00 @sgugger @fmassa. We might consider discussing this in a separate thread.

**Comment by jph00** [comment]
FYI `fastai2.medical.imaging.mask2bbox` does something fairly similar to what's requested in this issue. Although it's not a rotated box, and it uses a mask as input (use `torch.nonzero` to turn this into a list of points).

https://github.com/fastai/fastai_dev/blob/master/dev/local/medical/imaging.py#L175

**Comment by lebionick** [comment]
I wonder if it is possible to make differentiable crops based on tensor values (e.g. a neural network segments an image then we find bounding boxes in segmentation map and crop them from original image/feature map, concatenate and send to the next network)

**Comment by edgarriba** [comment]
@lebionick I believe this is possible with `crop_and_resize`
https://kornia.readthedocs.io/en/latest/geometry.transform.html#kornia.geometry.transform.crop_and_resize

analogous to ROI[Pool/Align] family operators for object detection algorithms

**Comment by lebionick** [comment]
@edgarriba
so the problem is to extract coordinates of bounding boxes for this function if doing segmentation, not based on sliding-window detection

**Comment by ducha-aiki** [comment]
@lebionick we are thinking about adding such function, thanks for suggestion.
In meantime you could you use this function from fastai2 (mask2bbox)
https://github.com/fastai/fastai2/blob/afed3afe7660a17a25c17f630f746918a4b11acb/fastai2/medical/imaging.py#L181

**Comment by lebionick** [comment]
@ducha-aiki 
As far as I understand, this function extracts only one box without angle. But for my case (multiline OCR) I would like to have many boxes and their angles.
P.S. The source code is a bit messy, I cannot understand how it works and how to extrapolate to my case:)

**Comment by jph00** [comment]
Please don't make judgements like that. Just because you can't understand it, doesn't make it messy. There might be other reasons you can't understand it. For instance, you don't have the context for that code, and haven't read the tutorial it's part of.

**Comment by domef** [comment]
any update?

**Comment by edgarriba** [comment]
Not in priority list. Feel free to propose a PR
